### PR TITLE
Support regex column searches when using CollectionEngine

### PR DESCRIPTION
**Pull request by cadenzajon**
The QueryBuilderEngine has support for using Oracle queries to search for regular expressions when a columns 'regex' flag is enabled. This update adds similar functionality to the CollectionEngine using PHP's built-in preg functions.

### CODE REVIEW

File: src/Engines/CollectionEngine.php [hunk 1]
```diff
@@ -177,20 +177,29 @@ public function columnSearch()
...
                 $this->collection = $this->collection->filter(
-                    function ($row) use ($column, $keyword) {
+                    function ($row) use ($column, $keyword, $regex) {
                         $data = $this->serialize($row);
 
                         $value = Arr::get($data, $column);
 
                         if ($this->isCaseInsensitive()) {
-                            return strpos(Str::lower($value), Str::lower($keyword)) !== false;
+                            if($regex) {
```
Comment (yajra) [inline src/Engines/CollectionEngine.php]: Please add a space after `if ()`. Thanks!

File: src/Engines/CollectionEngine.php [hunk 2]
```diff
@@ -177,20 +177,29 @@ public function columnSearch()
...
+                                return preg_match('/' . Str::lower($keyword) . '/', Str::lower($value)) == 1;
```
Comment (yajra) [inline src/Engines/CollectionEngine.php]: I think we can use preg_match with `i` flag as per this [SO post](http://stackoverflow.com/questions/12411037/how-do-i-make-this-preg-match-case-insensitive)?

**Comment by cadenzajon** [comment]
Thanks, I have made those updates!

**Comment by yajra** [comment]
Released on v6.19.0, thanks a lot! 👍

**Pull request merged**